### PR TITLE
MM-49603 : Remove fetching of deleted channels on page load

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- channel request types removed from fetchMyChannelsAndMembersREST
- removed `isMinimumServerVersion` check from that above action call, which is adding the query to include the archived channels

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49603

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: packages/mattermost-redux/src/actions/channels.ts [hunk 1]
```diff
@@ -460,52 +459,33 @@ export function getChannelTimezones(channelId: string): ActionFunc {
     };
 }
 
-export function fetchMyChannelsAndMembersREST(teamId: string): ActionFunc {
+export function fetchMyChannelsAndMembersREST(teamId: string): ActionFunc<{channels: ServerChannel[]; channelMembers: ChannelMembership[]}> {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        dispatch({
-            type: ChannelTypes.CHANNELS_REQUEST,
-            data: null,
-        });
```
Comment (M-ZubairAhmed) [inline packages/mattermost-redux/src/actions/channels.ts:468]: we will be getting rid of these

File: packages/mattermost-redux/src/actions/channels.ts [hunk 2]
```diff
@@ -516,7 +496,7 @@ export function fetchMyChannelsAndMembersREST(teamId: string): ActionFunc {
             dispatch(loadRolesIfNeeded(roles));
         }
 
-        return {data: {channels, members: channelMembers}};
+        return {data: {channels, channelMembers}};
```
Comment (AshishDhama) [inline packages/mattermost-redux/src/actions/channels.ts:499]: @M-ZubairAhmed is this key change not affecting any place we are using this function?

Comment (M-ZubairAhmed) [reply packages/mattermost-redux/src/actions/channels.ts:499]: i have checked the usage of this function, we are not using this actions return in any place, the reason for this change was to make it same as that of its gQL counterpart fetchChannelAndMembers

**Review by AshishDhama — COMMENTED** [review]
LGTM!

**Review by AshishDhama — APPROVED** [review]

**Comment by hmhealey** [comment]
Are you sure this isn't still needed? Archived channels can still be accessible in the UI depending on the server settings, so I'd think we'd still need them.

Isn't this action also used to resync after the websocket reconnects? It's possible that we'd need it to mark a channel as deleted if it was deleted while the websocket was disconnected

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.

**Comment by lieut-data** [comment]
@M-ZubairAhmed, can you re-open this as a PR against https://github.com/mattermost/mattermost-server/tree/mono-repo? Any QA review we do here may not be valid after we make the mono-repo live: currently targeting Wednesday.

**Pull request closed without merging**